packages/edit-site/src/components/header/index.js

```diff
@@ -79,64 +79,81 @@ export default function Header( {
 	const displayBlockToolbar =
 		! isLargeViewport || deviceType !== 'Desktop' || hasFixedToolbar;
 
+	const { hoveredEntityIds, getBlock } = useSelect( ( select ) => {
+		const { getHoveredEntities, getBlock } = select( 'core/block-editor' );
+
+		return {
+			hoveredEntityIds: getHoveredEntities(),
+			getBlock,
+		};
+	} );
+
+	const hoveredName =
+		getBlock( hoveredEntityIds[ 0 ] )?.attributes.slug || '';
+
```

Comment:
This is all that was changed in this file, please disregard the extra linting changes below.

Similarly, we would try to update this to show the title instead of the slug where available.